x-pack/test/apm_api_integration/tests/alerts/helpers/alerting_api_helper.ts

```diff
@@ -0,0 +1,234 @@
+}) {
+  const timerange = getTimerange();
+  const alertsCountReponse = await apmApiClient.readUser({
+    endpoint: 'GET /internal/apm/services/{serviceName}/alerts_count',
+    params: {
+      path: {
+        serviceName,
+      },
+      query: {
+        ...timerange,
+        environment: 'ENVIRONMENT_ALL',
+      },
+    },
+  });
+
+  return alertsCountReponse.body.alertsCount;
+}
+
+export async function runRuleSoon({
+  ruleId,
+  supertest,
+}: {
+  ruleId: string;
+  supertest: SuperTest<Test>;
+}): Promise<Record<string, any>> {
+  return pRetry(
+    async () => {
+      try {
+        const response = await supertest
+          .post(`/internal/alerting/rule/${ruleId}/_run_soon`)
+          .set('kbn-xsrf', 'foo');
+        // Sometimes the rule may already be running, which returns a 200. Try until it isn't
+        if (response.status !== 204) {
+          throw new Error(`runRuleSoon got ${response.status} status`);
+        }
+        return response;
+      } catch (error) {
+        throw new Error(`[Rule] Running a rule ${ruleId} failed: ${error}`);
+      }
+    },
+    { retries: 10 }
+  );
+}
+
+export async function deleteAlertsByRuleId({ es, ruleId }: { es: Client; ruleId: string }) {
+  await es.deleteByQuery({
+    index: APM_ALERTS_INDEX,
+    query: { term: { 'kibana.alert.rule.uuid': ruleId } },
+  });
+}
+
+export async function deleteRuleById({
+  supertest,
+  ruleId,
+}: {
+  supertest: SuperTest<Test>;
+  ruleId: string;
+}) {
+  await supertest.delete(`/api/alerting/rule/${ruleId}`).set('kbn-xsrf', 'foo');
+}
+
+export async function deleteApmRules(supertest: SuperTest<Test>) {
+  const res = await supertest.get(
+    `/api/alerting/rules/_find?filter=alert.attributes.consumer:apm&per_page=10000`
+  );
+
+  return Promise.all(
+    res.body.data.map(async (rule: any) => {
+      await supertest.delete(`/api/alerting/rule/${rule.id}`).set('kbn-xsrf', 'foo');
+    })
+  );
+}
+
+export function deleteApmAlerts(es: Client) {
+  return es.deleteByQuery({ index: APM_ALERTS_INDEX, query: { match_all: {} } });
+}
+
+export async function clearKibanaApmEventLog(es: Client) {
+  return es.deleteByQuery({
+    index: '.kibana-event-log-*',
+    query: { term: { 'kibana.alert.rule.consumer': 'apm' } },
+  });
+}
+
+export type ApmAlertFields = ParsedTechnicalFields & {
+  'service.name': string;
+  'service.environment': string;
+  'transaction.name': string;
+  'error.grouping_key': string;
+  'error.grouping_name': string;
+};
+
+export async function getAlertByRuleId({ es, ruleId }: { es: Client; ruleId: string }) {
+  const response = (await es.search({
+    index: APM_ALERTS_INDEX,
+    body: {
+      query: {
+        term: {
+          'kibana.alert.rule.uuid': ruleId,
+        },
+      },
+    },
+  })) as SearchResponse<ApmAlertFields, Record<string, AggregationsAggregate>>;
+
+  return response.hits.hits.map((hit) => hit._source) as ApmAlertFields[];
+}
+
+export async function getActiveApmAlerts({
```

Comment:
same here, is used only in waitForActiveApmAlert

Reply:
Fixed